pkg/kubernetes/kube.go

```diff
@@ -42,6 +42,7 @@ type Client struct {
 	DynamicInterface   dynamic.Interface
 	DiscoveryInterface discovery.DiscoveryInterface
 	FilesPath          string
+	TemplateArguments  interface{}
```

Comment:
How do we plan to have the user interact with this interface? I know we want to give the user the flexibility to define the templated and environment variables, but are they also going to code the template struct, and the gathering of the environment variables?

I ask because when I think about them doing the gathering of the environment variables, I can't think of a good place to do it. Maybe we could do a little wrapping where the user only has to call a simple method to add the names for the templated and environment variables and we do the gathering internally. What do you think?

Reply:
Users can configure kubedog client:
```
t.TemplateArguments = myDataStruct
```

Creating and passing that struct is still the job of who ever is running the test.
If you feel it's more appropriate we can have a setter method to do this instead, e.g.
```
t.SetTemplateArguments(myDataStruct)
```

Reply:
I guess we can do that later if we see it fit